### PR TITLE
package: uses jshint and mocha, added as dev-deps

### PR DESCRIPTION
**Pull request by sam-github**
@bajtos Can I pass this to you to make the tests pass (or mark as skipped if you feel irrelevant).

I fixed the obvious problems, the missing dev dependencies, and the jshint regression: https://github.com/jshint/jshint/issues/1747

### CODE REVIEW

**Comment by bajtos** [comment]
I don't know when I'll have time to work on this, but I'll try to find some time in the next few weeks.

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "sl-registry": "bin/sl-registry.js"
   },
   "scripts": {
-    "pretest": "jshint .",
+    "lint": "jshint .",
```
Comment (bajtos) [inline package.json]: Is `npm test` running the `lint` script? If not, then this is effectively removing code-style check from `npm test` :(

Comment (sam-github) [inline package.json]: I think its annoying to have lint be a pre-req of 'test', myself. I do `npm run lint` from my git pre-commit hook, but feel free to add it back in if you like it. Note that the pretest result is ignored by CI, so its purely a convenience for those who like lint to happen at time of `npm test`.

Comment (bajtos) [inline package.json]: FWIW, in https://github.com/strongloop/strong-remoting/pull/195 I am pushing for not ignoring the result of `pretest` script on CI.

**Comment by bajtos** [comment]
ok, I'm taking this over.

**Comment by bajtos** [comment]
@slnode test please

**Comment by bajtos** [comment]
@slnode test please

**Comment by bajtos** [comment]
@sam-github PTAL, all is green now.

**Comment by sam-github** [comment]
LGTM

**Pull request merged**